core/modules/comment/comment.module

```diff
@@ -935,55 +935,8 @@ function comment_prepare_thread(&$comments) {
  *   An array as expected by backdrop_render().
  */
 function comment_view(Comment $comment, Node $node, $view_mode = 'full', $langcode = NULL) {
-  if (!isset($langcode)) {
-    $langcode = $GLOBALS['language_content']->langcode;
-  }
-
-  // Populate $comment->content with a render() array.
-  comment_build_content($comment, $node, $view_mode, $langcode);
-
-  $build = $comment->content;
-  // We don't need duplicate rendering info in comment->content.
-  unset($comment->content);
-
-  $build += array(
-    '#theme' => 'comment__node_' . $node->type . '__' . $view_mode,
-    '#comment' => $comment,
-    '#node' => $node,
-    '#view_mode' => $view_mode,
-    '#language' => $langcode,
-  );
-  $node_type = node_type_get_type($node->type);
-
-  if (empty($comment->in_preview)) {
-    $prefix = '';
-    $is_threaded = isset($comment->divs) && $node_type->settings['comment_mode'] == COMMENT_MODE_THREADED;
-
-    // Add 'new' anchor if needed.
-    if (!empty($comment->first_new)) {
-      $prefix .= "<a id=\"new\"></a>\n";
-    }
-
-    // Add indentation div or close open divs as needed.
-    if ($is_threaded) {
-      $prefix .= $comment->divs <= 0 ? str_repeat('</div>', abs($comment->divs)) : "\n" . '<div class="indented">';
-    }
-
-    // Add anchor for each comment.
-    $prefix .= "<a id=\"comment-$comment->cid\"></a>\n";
-    $build['#prefix'] = $prefix;
-
-    // Close all open divs.
-    if ($is_threaded && !empty($comment->divs_final)) {
-      $build['#suffix'] = str_repeat('</div>', $comment->divs_final);
-    }
-  }
-
-  // Allow modules to modify the structured comment.
-  $type = 'comment';
-  backdrop_alter(array('comment_view', 'entity_view'), $build, $type);
-
-  return $build;
+  $view = entity_get_controller('comment')->view(array($comment->cid => $comment), $view_mode, $langcode);
```

Comment:
In this and all subsequent `*_view()` wrappers, how would you feel about using `return $comment->view()` instead? Seems like more expected content for a wrapper, even though `$comment->view()` is ultimately going to call the same thing (so the current solution is less nesting and slightly more efficient).

Reply:
I think it makes it a bit neater, sure.

Reply:
@quicksketch I've made these changes and all tests are passing.